workflows/index-generation/generate_ncbi_db_index.py

```diff
@@ -0,0 +1,85 @@
+"""
+Generate loc db
+Generate Loc DB for NT/NR
+"""
+import sys
+import logging
+from collections import namedtuple
+
+import marisa_trie
+
+
+MILLION = 1_000_000
+
+
+def _accession_id_and_name(header_line):
+    # Sometimes multiple accessions will be mapped to a single sequence.
+    # In this case, they will be separated by the \x01 char.
+    # To get the accession name, just use the string until the first \x01.
+    parts = header_line.split('\x01', 1)[0].split(' ', 1)
+    accession_id = parts[0][1:]  # Remove the '>'
+    accession_name = parts[1] if len(parts) > 1 else ''
+    return accession_id, accession_name
+
+
+Accession = namedtuple('Accession', 'accession_id accession_name seq_offset header_len seq_len seq_bp_len')
```

Comment:
nit: I like the syntax where the field names are a list instead of a string
```suggestion
Accession = namedtuple('Accession', ['accession_id', 'accession_name', 'seq_offset', 'header_len', 'seq_len', 'seq_bp_len'])
```